src/cli/cmd/mod.rs

```diff
@@ -477,6 +482,22 @@ fn validate_segment(s: &str) -> Result<(), FhError> {
     Ok(())
 }
 
+// See if the netrc exists at the /nix/var/determinate/netrc and, if not, try
+// to find it via XDG path.
+async fn get_netrc_path(xdg: &BaseDirectories) -> Result<PathBuf, FhError> {
```

Comment:
I kinda wonder about this... I think we probably need two separate functions: one for when we're getting a path-to-write-to (i.e. the user running `fh login` without determinate-nixd running), and one for when we're getting a path-to-read-from.

In the write-to one, it would take into account if determinate-nixd is running and return `None` if it is (since it won't write to a file, but will instead tell the daemon) and `Some(xdg-path)` if it's not.

In the read-from one, it would return netrcnotfound if the determinate and normal netrc files are not found.

Reply:
I think the `dnixd_uds` function is a good starting point, but basically request to `http://localhost/info` and see if it responds with 200.

Reply:
@cole-h What action do we take if the write-to path is `None`?

Reply:
I imagine that case is only reachable when we're using dnixd, so we just wouldn't write a file and tell the daemon to use this token the same way we do it already.